src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -124,6 +122,28 @@ public partial struct SyntaxValueProvider
         }
     }
 
+    private record SyntaxTreeInfo(SyntaxTree Tree, bool ContainsGlobalAliases, bool ContainsAttributeList);
+    private static readonly ConditionalWeakTable<SyntaxTree, SyntaxTreeInfo> s_treeToInfo = new();
+
+    private static SyntaxTreeInfo GetTreeInfo(
+        SyntaxTree tree, ISyntaxHelper syntaxHelper, CancellationToken cancellationToken)
+    {
+        // prevent captures for the case where the item is in the tree.
+        return s_treeToInfo.TryGetValue(tree, out var info)
+            ? info
+            : computeTreeInfo();
```

Comment:
critically, we compute and cache the data about the syntax-tree in a CWT.  The reason for this is that on every edit we're gonig to get called back with N trees from teh compilation, N-1 of which are identical to before.  We don't want to recompute the information for each one to determine if we should filter it out.  THe info will only last as long as the tree itself lasts.  Once it gets GC'ed, this gets dropped.